src/iam/certhandler/modules/pkcs11.cpp

```diff
@@ -0,0 +1,986 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/modules/pkcs11.hpp"
+
+#include "aos/common/tools/fs.hpp"
+#include "aos/common/tools/os.hpp"
+#include "aos/common/uuid.hpp"
+
+#include "../log.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/**
+ * A helper object for search operations containing the most valuable data for certhandler.
+ */
+struct SearchObject {
+    /**
+     * Object type.
+     */
+    Optional<pkcs11::ObjectClass> mType;
+    /**
+     * Object handle.
+     */
+    pkcs11::ObjectHandle mHandle;
+    /**
+     * Object label.
+     */
+    StaticString<pkcs11::cLabelLen> mLabel;
+    /**
+     * Key identifier for public/private key pair.
+     */
+    uuid::UUID mID;
+};
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+PKCS11Module::PKCS11Module(const String& certType, const ModuleConfig& config)
+    : mCertType(certType)
+    , mConfig(config)
+{
+}
+
+Error PKCS11Module::Init(
+    pkcs11::LibraryContext& pkcs11, crypto::x509::ProviderItf& x509Provider, uuid::UUIDManagerItf& uuidManager)
+{
+    mPKCS11 = &pkcs11;
+    mX509Provider = &x509Provider;
+    mUUIDManager = &uuidManager;
+
+    Error err = Error::Enum::eNone;
```

Comment:
`Error` is initialized with none: `Error err;` should be ok. But explicit initialization is ok too.